spec/libraries/duplicator_spec.rb

```diff
@@ -0,0 +1,763 @@
+require 'rails_helper'
+
+RSpec.describe Duplicator, type: :model do
+  context 'when Plain Old Ruby Objects' do
+    class SimpleObject
+      def initialize(id)
+        @id = id
+      end
+
+      def initialize_duplicate(_duplicator, other)
+      end
+
+      def ==(other)
+        self.class == other.class && state == other.state
+      end
+
+      protected
+
+      def state
+        [@id]
+      end
+    end
+
+    # ComplexObject has children
+    class ComplexObject
+      attr_reader :children
+
+      def initialize(id, children)
+        @id = id
+        @children = children
+      end
+
+      def initialize_duplicate(duplicator, other)
```

Comment:
Unused method argument - `other`. If it's necessary, use `_` or `_other` as an argument name to indicate that it won't be used.